File: src/main/java/it/flowing/workshop/controllers/UsersController.java
package it.flowing.workshop.controllers;

import it.flowing.workshop.model.User;
import it.flowing.workshop.services.UserService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import java.util.List;

import static it.flowing.workshop.ApiVersion.V1;

@RestController
@RequestMapping(value = "/" + V1 + "/users")
public class UsersController {

    private final UserService userService;

    @Autowired
    public UsersController(UserService userService) {
        this.userService = userService;
    }

    @GetMapping
    @SkipAuthentication
    public List<User> list() {
        return userService.list();
    }

    @GetMapping("/{id}")
    @SkipAuthentication
    public User get(@PathVariable("id") String id) {
        return userService.get(id);
    }

    @PostMapping
    public User create(@RequestBody User toInsert) {
        return userService.insert(toInsert);
    }

    @PutMapping("/{id}")
    public User update(@RequestBody User toUpdate, @PathVariable("id") String id) {
        return userService.update(id, toUpdate);
    }

    @DeleteMapping("/{id}")
    public void delete(@PathVariable("id") String id) {
        userService.delete(id);
    }
}

File: src/main/java/it/flowing/workshop/controllers/UsersController.kt
package it.flowing.workshop.controllers

import it.flowing.workshop.ApiVersion
import it.flowing.workshop.model.User
import it.flowing.workshop.services.UserService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping(value = ["/" + ApiVersion.V1 + "/users"])
class UsersController @Autowired constructor(private val userService: UserService) {
    @GetMapping
    @SkipAuthentication
    fun list(): List<User> {
        return userService.list()
    }

    @GetMapping("/{id}")
    @SkipAuthentication
    operator fun get(@PathVariable("id") id: String): User {
        return userService[id]
    }

    @PostMapping
    fun create(@RequestBody toInsert: User): User {
        return userService.insert(toInsert)
    }

    @PutMapping("/{id}")
    fun update(@RequestBody toUpdate: User, @PathVariable("id") id: String): User {
        return userService.update(id, toUpdate)
    }

    @DeleteMapping("/{id}")
    fun delete(@PathVariable("id") id: String) {
        userService.delete(id)
    }

}